src/files/manager.ts

```diff
@@ -229,4 +237,114 @@ export class FilesManager {
       throw error;
     }
   }
+
+  /**
+   * Uploads a new media file to the Strapi Media Library.
+   *
+   * @param file - The file to upload (Blob or Buffer).
+   * @param options - Upload options. For Buffer uploads, filename and mimetype are required.
+   * @returns A promise that resolves to the uploaded file information.
+   *
+   * @throws {FileForbiddenError} if the user does not have permission to upload files.
+   * @throws {HTTPError} if the HTTP client encounters connection issues, the server is unreachable, or authentication fails.
+   *
+   * @example
+   * ```typescript
+   * const filesManager = new FilesManager(httpClient);
+   *
+   * // Upload with a File object (browser)
+   * const fileInput = document.querySelector('input[type="file"]');
+   * const file = fileInput.files[0];
+   * const result = await filesManager.upload(file, {
+   *   fileInfo: { alternativeText: 'A user uploaded image' }
+   * });
+   *
+   * // Upload with a Blob
+   * import { blobFrom } from 'node-fetch';
+   * const file = await blobFrom('./1.png', 'image/png');
+   * const result = await filesManager.upload(file, {
+   *   fileInfo: { alternativeText: 'An example image' }
+   * });
+   *
+   * // Upload with a Buffer (Node.js) - filename and mimetype are required
+   * import fs from 'fs';
+   * const buffer = fs.readFileSync('./image.png');
+   * const result = await filesManager.upload(buffer, {
+   *   filename: 'image.png',
+   *   mimetype: 'image/png',
+   *   fileInfo: { alternativeText: 'An example image' }
+   * });
+   *
+   * console.log(result); // Upload response with file details
+   * ```
+   */
+  async upload(file: Blob, options?: BlobUploadOptions): Promise<MediaUploadResponse>;
+  async upload(file: Buffer, options: BufferUploadOptions): Promise<MediaUploadResponse>;
+  async upload(
+    file: Blob | Buffer,
+    options?: BlobUploadOptions | BufferUploadOptions
+  ): Promise<MediaUploadResponse> {
+    debug('uploading new file');
+
+    try {
+      const url = FILE_API_PREFIX;
+      const client = this.createFileHttpClient();
+
+      if (typeof Buffer !== 'undefined' && Buffer.isBuffer(file)) {
+        if (!options || !('filename' in options) || !('mimetype' in options)) {
+          throw new Error('Filename and mimetype are required when uploading Buffer data');
+        }
+
+        return this.uploadBuffer(file, options as BufferUploadOptions, client, url);
+      } else {
```

Comment:
```typescript
      if (file instanceof Buffer && typeof Buffer !== 'undefined' && Buffer.isBuffer(file)) {
        if (!options || !('filename' in options) || !('mimetype' in options)) {
          throw new Error('Filename and mimetype are required when uploading Buffer data');
        }

        return this.uploadBuffer(file, options, client, url);
      } else if (file instanceof Blob) {
        return this.uploadBlob(file, options, client, url);
      } else {
        console.warn('Could not determine type of file; attempting upload as Blob');
        return this.uploadBlob(file as unknown as Blob, options as BlobUploadOptions, client, url);
      }
```

Haven't tested it, but maybe something like this just to be sure and help with debugging